### PR TITLE
Support device specifiers in pix2pix example 

### PR DESCRIPTION
**Pull request by hvy**
Part of https://github.com/chainer/chainer/issues/6661.

Needs
- `unchain_backward`, similar to #7055.

### CODE REVIEW

File: examples/pix2pix/train_facade.py [hunk 1]
```diff
@@ -39,23 +44,32 @@ def main():
...
+                       help='GPU ID (negative value indicates CPU)')
     args = parser.parse_args()
 
-    print('GPU: {}'.format(args.gpu))
+    device = chainer.get_device(args.device)
+    if device.xp is chainerx:
+        sys.stderr.write('This example does not support ChainerX devices.\n')
+        sys.exit(1)
+
+    print('GPU: {}'.format(device))
```
Comment (niboshi) [inline examples/pix2pix/train_facade.py]: How about "Device"?

Comment (hvy) [reply examples/pix2pix/train_facade.py]: Thanks, fixed.

**Comment by hvy** [comment]
PTAL.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**